### ISSUE TITLE
Provide objects and helpers in Nitro plugins

### ISSUE THREAD

**Issue by cybercoder-naj** [unjs/nitro#2322]
### Describe the feature

I need to create a discord bot in Nitro (with Nuxt 3). In a simple node application, I can use `src/index.js` and easily setup my bot and make it online. However, the docs make it seem like the only thing Nitro can do is make API endpoints.

When my Nuxt application is deployed, I want to execute this code:
```js
// Require the necessary discord.js classes
const { Client, Events, GatewayIntentBits } = require('discord.js');
const { token } = require('./config.json');

// Create a new client instance
const client = new Client({ intents: [GatewayIntentBits.Guilds] });

// When the client is ready, run this code (only once).
// The distinction between `client: Client<boolean>` and `readyClient: Client<true>` is important for TypeScript developers.
// It makes some properties non-nullable.
client.once(Events.ClientReady, readyClient => {
	console.log(`Ready! Logged in as ${readyClient.user.tag}`);
});

// Log in to Discord with your client's token
client.login(token);
```

But I have no idea where to put this. 

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by cybercoder-naj** [comment]
Re: I realised I need to register a plugin. But it was after days of going through the docs multiple times. The docs need to be restructured such that some information is displayed for easier access. Nitro is great except many people suggest that the docs need a lot of improvement. 

Additionally, in Nuxt, I can attach some objects to the `nuxtApp` in a plugin, clearly mentioned in the nuxt docs [here](https://nuxt.com/docs/guide/directory-structure/plugins#providing-helpers). I would love such a functionality in Nitro as well.

**Comment by cybercoder-naj** [comment]
Closing as duplicate of #711
